### PR TITLE
refactor: Refactor pyrorisks

### PR DESCRIPTION
**Pull request by jsakv**
The PR introduces the following changes:

- refactor: 
  - Remove Heroku deployment files; the service is no longer used by Pyronear
  - Remove CML / DVC pipelines; will revisit mlops tooling later if necessary  
  - Deprecate the outdated pyro_risks modules, CLI, and examples

- chore: 
  - Add poetry for packaging and dependency management
  - Add ruff for linting and formatting
  - Update docker build and config
  - Update CI workflows
  - Bump Geopandas to 1.0.1, replace fiona backend with pyogrio with speedups >5-20x   

- style: Fix type hints, formatting, and linting 
- docs: Remove deprecated modules from docs

To speed up the API deployment, the refactored version of the repository only targets Linux distributions and python 3.10; multi-platform and multi-version builds will come later. It will also allow us to assess if  a mix of poetry and conda is required for installation across platforms (cf `pyrovision`, `pyroengine`). 

I plan on adding unit test workflow with the next PR!

Any feedback is welcomed 🌲

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+[tool.poetry.group.dev.dependencies]
+pytest = "^7.4.2"
+black = "^23.9.1"
+pre-commit = "^3.4.0"
+mypy = "^1.10.0"
+ruff = "^0.5.1"
+pytest-cov = "^5.0.0"
+
+
+[tool.poetry.group.app.dependencies]
```
Comment (Acruve15) [inline pyproject.toml:47]: [QUESTION] should we name it app or api?

**Review by Acruve15 — APPROVED** [review]

File: pyproject.toml
Comment (juldpnt) [inline pyproject.toml:1]: [QUESTION] Why use both ruff and black ? Ruff can be a direct replacement for black, so it seems it is redundant

File: pyproject.toml (hunk 1 above)
Comment (jsakv) [reply pyproject.toml:47]: Both can work. I named it `app` to match the top-level directory and the fast API app/API convention, but `api` is straightforward!

File: pyproject.toml
Comment (jsakv) [reply pyproject.toml:1]: True, thanks for catching that!

**Review by juldpnt — COMMENTED** [review]
Thank you for this PR !! I have a little question, I'm wondering if you chose Python 3.10 for technical reasons/compatibility? The 3.11 (or even 3.12) would give many QOLs for typing, interpreter errors, etc.

**Comment by jsakv** [comment]
> Thank you for this PR !! I have a little question, I'm wondering if you chose Python 3.10 for technical reasons/compatibility? The 3.11 (or even 3.12) would give many QOLs for typing, interpreter errors, etc.

Fair question; the main concern was to avoid incompatibilities and reduce the surface for build errors. 
For instance, geopandas supports Python 3.9+, but rasterio has 3.12 wheels but is not being tested (CI) on Python 3.12, and pyogrio has had compatibility [issues](https://github.com/geopandas/pyogrio/issues/416) recently.

These few points are enough to rule out `Python 3.12 preventively` IMHO, and on the other side`Python 3.9` is approaching its end of life in ~ one year. 

The package is compatible with both Python 3.10 and 3.11, but for the docker image and the CI, I went with the 3.10 version. The idea is to introduce relevant multi-platform and multi-version builds later!

**Review by juldpnt — APPROVED** [review]

**Pull request merged**